src/buffer/out/textBuffer.cpp

```diff
@@ -1400,26 +1400,29 @@ void TextBuffer::_ExpandTextRow(SMALL_RECT& textRow) const
 // Routine Description:
 // - Retrieves the text data from the selected region and presents it in a clipboard-ready format (given little post-processing).
 // Arguments:
-// - lineSelection - true if entire line is being selected. False otherwise (box selection)
+// - lineSelection - true if entire line is being selected. False otherwise (block selection)
 // - trimTrailingWhitespace - setting flag removes trailing whitespace at the end of each row in selection
-// - selectionRects - the selection regions from which the data will be extracted from the buffer
-// - GetForegroundColor - function used to map TextAttribute to RGB COLORREF for foreground color
-// - GetBackgroundColor - function used to map TextAttribute to RGB COLORREF for foreground color
+// - textRects - the rectangular regions from which the data will be extracted from the buffer (i.e.: selection rects)
+// - GetForegroundColor - function used to map TextAttribute to RGB COLORREF for foreground color. If null, only extract the text.
+// - GetBackgroundColor - function used to map TextAttribute to RGB COLORREF for background color. If null, only extract the text.
 // Return Value:
 // - The text, background color, and foreground color data of the selected region of the text buffer.
-const TextBuffer::TextAndColor TextBuffer::GetTextForClipboard(const bool lineSelection,
-                                                               const bool trimTrailingWhitespace,
-                                                               const std::vector<SMALL_RECT>& selectionRects,
-                                                               std::function<COLORREF(TextAttribute&)> GetForegroundColor,
-                                                               std::function<COLORREF(TextAttribute&)> GetBackgroundColor) const
+const TextBuffer::TextAndColor TextBuffer::GetText(const bool includeCRLF,
+                                                   const bool trimTrailingWhitespace,
+                                                   const std::vector<SMALL_RECT>& selectionRects,
+                                                   std::function<COLORREF(TextAttribute&)> GetForegroundColor,
+                                                   std::function<COLORREF(TextAttribute&)> GetBackgroundColor) const
 {
     TextAndColor data;
 
     // preallocate our vectors to reduce reallocs
     size_t const rows = selectionRects.size();
     data.text.reserve(rows);
-    data.FgAttr.reserve(rows);
-    data.BkAttr.reserve(rows);
+    if (GetForegroundColor && GetBackgroundColor)
+    {
+        data.FgAttr.reserve(rows);
+        data.BkAttr.reserve(rows);
+    }

```

Comment:
What if I give you one and not the other? This is weird given that I can choose to do that. (And below)

Reply:
Right now, we don't have a use case for extracting only one of the attributes. So, you either get text with foreground+background, or just text with no data for foreground/background.

Reply:
Sure, but you're setting up someone for a mistake. If the two functions must be provided together, they should come in as a pair. If it is possible for each one to be set separately, then we should test each one for reserve/filling. Your choice.